### PR TITLE
[ZEPPELIN-5468]Fast return when invalid ticket of no session case

### PR DESCRIPTION
**Pull request by aib628**
### What is this PR for?
Fix NullPointerException of org.apache.zeppelin.socket.NotebookServer when invalid ticket message received 
This exception may occur when server sessions is not contains received principal. In other words, it happend when no session case.

### What type of PR is it?
Bug Fix

### Todos
haven't

### What is the Jira issue?
[ZEPPELIN-5468] Fix NullPointerException of org.apache.zeppelin.socket.NotebookServer when invalid ticket message received

### How should this be tested?
Login visit page and then logout in other page, see log output

### Screenshots (if appropriate)
image

### Questions:
* Does the licenses files need update?  no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

**Comment by zjffdu** [comment]
@aib628 CI is failed, could you rebase and trigger it again ? And it would better to create PR against master branch

**Comment by aib628** [comment]
> @aib628 CI is failed, could you rebase and trigger it again ? And it would better to create PR against master branch

I had rebase and trigger it again, but CI still failed. How can i rerun and trigger it agagin when rebase branch no update?

**Comment by zjffdu** [comment]
@aib628 Just force push will trigger the CI, and BTW besides the NPE in log, is there any user experience effect of this issue ??

**Comment by aib628** [comment]
> @aib628 Just force push will trigger the CI, and BTW besides the NPE in log, is there any user experience effect of this issue ??

@zjffdu no, log only

**Comment by zjffdu** [comment]
@aib628  Recently CI seems unable recently, we may need to wait for a while until CI become stable.

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java [hunk 1]
```diff
@@ -263,19 +263,17 @@ public void onMessage(NotebookSocket conn, String msg) {
...
         if (StringUtils.isEmpty(receivedMessage.ticket)) {
-          LOG.debug("{} message: invalid ticket {} != {}", receivedMessage.op,
-                  receivedMessage.ticket, ticketEntry.getTicket());
-        } else {
-          if (!receivedMessage.op.equals(OP.PING)) {
-            conn.send(serializeMessage(new Message(OP.SESSION_LOGOUT).put("info",
-                    "Your ticket is invalid possibly due to server restart. "
-                            + "Please login again.")));
-          }
+          LOG.debug("{} message: invalid ticket {} != {}", receivedMessage.op, receivedMessage.ticket, ticketEntry.getTicket());
```
Comment (Reamer) [inline zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:272]: `receivedMessage.ticket` is empty or null at this time. I think that this message makes no sense.

Comment (aib628) [reply zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:272]: Yeah, i'm with you on that. But it appears, as shown in the screenshot, so we log it and fast return, other than execute downstram logic, and then resulting in NPE.

Comment (Reamer) [reply zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:272]: Your screenshot shows a message that is logged at [line 466](https://github.com/apache/zeppelin/blob/eb0f8ab8c07c83f054a9de23aaa4edf9239e585c/zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java#L466).
The debug log level is turned off by default. I think we should remove the first if and always log the line in the debug log level.

Comment (aib628) [reply zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:272]: I don't quite understand what you mean, but i think a invalid ticket is no sence, we shoud log it in debug log level and fast return.

Comment (Reamer) [reply zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:272]: I mean this:
```
} else if (!ticketEntry.getTicket().equals(receivedMessage.ticket)) {
  LOG.debug("{} message: invalid ticket {} != {}", receivedMessage.op, receivedMessage.ticket, ticketEntry.getTicket());
  if (!receivedMessage.op.equals(OP.PING)) {
    conn.send(serializeMessage(new Message(OP.SESSION_LOGOUT).put("info", "Your ticket is invalid possibly due to server restart. Please login again.")));
  }
  return;
}

Comment (aib628) [reply zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:272]: This is an existing code fragment, and it means send logout messages to client when invalid ticket, i think we can keep it.

Comment (aib628) [reply zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:272]: good proposal, let me update it, thx.

**Review by Reamer — APPROVED** [review]
LGTM

**Comment by Reamer** [comment]
If the CI goes through successfully, I will soon merge it into master.

**Pull request closed without merging**